src/pages/MainPage.vue

```diff
@@ -332,7 +329,7 @@ export default {
   mounted () {
     const today = new Date();
     const month = today.getMonth();
-    this.currentSemester = (month >= 4 && month <= 10) ? 1 : 3;
+    this.$store.state.currentSemester = (month >= 4 && month <= 10) ? 1 : 3;
```

Comment:
it's better to use a mutation here (`setCurrentSemester`), Vuex doesn't like it when state is directly modified externally